File: solutions/clojure/bob/1/bob.clj
(ns bob)

(use 'clojure.string)

;; question -> Sure.
;; yell -> Whoa, chill out!
;; without saying -> Fine. Be that way!
;; else -> Whatever.

(defn response-for
  [msg]
  (def response "Whatever.")
  (def is-question (= \? (last (seq msg))))
  (def all-upper 
    (re-matches #"[A-Z]+"
      (join "" (re-seq #"[a-z|A-Z]+" msg))))
  (def all-spaces (empty? (trim msg)))

  (if is-question (def response "Sure."))
  (if all-upper   (def response "Whoa, chill out!"))
  (if all-spaces  (def response "Fine. Be that way!"))
  response
)

File: solutions/csharp/bob/1/Bob.cs
public static class Bob
{
    public static string Response(string statement)
    {
        var trimmedStatement = statement.Trim();

        return (IsSilence(trimmedStatement), IsYelling(trimmedStatement), IsQuestion(trimmedStatement)) switch
        {
            (true, _, _) => "Fine. Be that way!",
            (_, true, true) => "Calm down, I know what I'm doing!",
            (_, true, false) => "Whoa, chill out!",
            (_, false, true) => "Sure.",
            _ => "Whatever."
        };
    }

    private static bool IsSilence(string statement) => statement.Length == 0;

    private static bool IsQuestion(string statement) => statement.EndsWith('?');

    private static bool IsYelling(string statement) =>
        statement.Any(char.IsLetter) && !statement.Any(char.IsLower);
}
